src/mainwindow.cpp

```diff
@@ -208,14 +208,13 @@ void MainWindow::loadUserSettings() {
 }
 
 void MainWindow::restoreWindowState() {
-    QSettings settings;
-
-    if (settings.contains("saved_widget_geometry")) {
-        this->restoreGeometry(settings.value("window_geometry").toByteArray());
-        this->restoreState(settings.value("window_state").toByteArray());
-        this->ui->splitter_map->restoreState(settings.value("map_splitter_state").toByteArray());
-        this->ui->splitter_events->restoreState(settings.value("events_splitter_state").toByteArray());
-        this->ui->splitter_main->restoreState(settings.value("main_splitter_state").toByteArray());
+    if (porymapConfig.getRestoreWindowGeometry()) {
+        logInfo("Restoring window geometry from previous session.");
```

Comment:
2 code improvements I'd like to see here:
1. `porymapConfig.getGeometry()` should return a `QMap`, rather than an array.
2. Call `getGeometry()` once, and use the result when restoring state.